### PR TITLE
produce correct enum for jaxrs resteasy eap

### PR DESCRIPTION
**Pull request by TheCK**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: master for non-breaking changes and `3.0.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

- fixed the mustache enum-template for java jaxrs-resteasy-eap code generation to produce valid java (see issue #4123)

### CODE REVIEW

**Comment by wing328** [comment]
Please run resteasy-related petstore scripts to update the samples so that CIs can verify the result:
```
swagger-codegen|master ⇒ ./bin/java-petstore-resteasy.sh
java-petstore-resteasy.sh*                    jaxrs-resteasy-eap-joda-petstore-server.sh*   jaxrs-resteasy-joda-petstore-server.sh*                                                   
jaxrs-resteasy-eap-java8-petstore-server.sh*  jaxrs-resteasy-eap-petstore-server.sh*        jaxrs-resteasy-petstore-server.sh* 
```

cc @bbdouglas @JFCote @sreeshas @jfiala @lukoyanov @ePaul @cbornet

**Comment by TheCK** [comment]
@wing328 I have done that. They did not change anything in the example, which is to be expected. If the example had used enums it would not have compiled, since it was broken

**Comment by wing328** [comment]
@TheCK Good. If you've time, please update the script to use https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml instead of petstore.yaml as the former covers a lot more edge cases (including enum) but that might result in other issue with the generated code so it's something optional (not required for this PR to be merged)

File: modules/swagger-codegen/src/main/resources/JavaJaxRS/resteasy/eap/enumOuterClass.mustache [hunk 1]
```diff
@@ -1,3 +1,7 @@
 public enum {{classname}} {
-    {{#allowableValues}}{{.}}{{^-last}}, {{/-last}}{{/allowableValues}}
+  {{#allowableValues}}
+  {{#enumVars}}
+  {{{name}}}{{^-last}},{{/-last}}{{#-last}};{{/-last}}
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/JavaJaxRS/resteasy/eap/enumOuterClass.mustache]: Minor suggestion: we might want to add a space after ","
